packages/kbn-config-schema/src/types/type.ts

```diff
@@ -88,30 +106,39 @@ export abstract class Type<V> {
     return undefined;
   }
 
-  private onError(error: SchemaTypeError | ValidationErrorItem): SchemaTypeError {
+  private onError(error: SchemaTypeError | ErrorReport): SchemaTypeError {
     if (error instanceof SchemaTypeError) {
       return error;
     }
 
-    const { context = {}, type, path, message } = error;
+    const { local, code, path, value } = error;
+    // `message` is only initialized once `toString` has been called (...)
+    // see https://github.com/sideway/joi/blob/master/lib/errors.js
+    const message = error.toString();
```

Comment:
nit: `message` is used at the very bottom. Maybe we can move it closer to the reading place?